extensions/hibernate-validator/deployment/src/main/java/io/quarkus/hibernate/validator/deployment/HibernateValidatorProcessor.java

```diff
@@ -327,6 +339,34 @@ private static void contributeMethodsWithInheritedValidation(Map<DotName, Set<St
         }
     }
 
+    private static Map<DotName, Set<String>> gatherJaxRsMethods(
+            List<AdditionalJaxRsResourceMethodAnnotationsBuildItem> additionalJaxRsResourceMethodAnnotations,
+            IndexView indexView) {
+        Map<DotName, Set<String>> jaxRsMethods = new HashMap<>();
+
+        Collection<DotName> effectiveJaxRsMethodDefiningAnnotations = new ArrayList<>(
+                JAXRS_METHOD_ANNOTATIONS.length + additionalJaxRsResourceMethodAnnotations.size());
+        effectiveJaxRsMethodDefiningAnnotations.addAll(Arrays.asList(JAXRS_METHOD_ANNOTATIONS));
+        for (AdditionalJaxRsResourceMethodAnnotationsBuildItem additionalJaxRsResourceMethodAnnotation : additionalJaxRsResourceMethodAnnotations) {
+            effectiveJaxRsMethodDefiningAnnotations.addAll(additionalJaxRsResourceMethodAnnotation.getAnnotationClasses());
+        }
+
+        for (DotName jaxRsAnnotation : effectiveJaxRsMethodDefiningAnnotations) {
+            Collection<AnnotationInstance> annotationInstances = indexView.getAnnotations(jaxRsAnnotation);
+
+            if (annotationInstances.isEmpty()) {
+                continue;
+            }
+
+            for (AnnotationInstance annotation : annotationInstances) {
+                MethodInfo method = annotation.target().asMethod();
+                jaxRsMethods.computeIfAbsent(method.declaringClass().name(), k -> new HashSet<String>())
+                        .add(method.name().toString());
```

Comment:
The name is not sufficient, you need to build the signature.

Reply:
Getting the `parameters()` then building something with the `name()` of the type should be good enough.